### ISSUE TITLE
Support multiple directories for spark.local.dir

### ISSUE THREAD

**Issue by weiting-chen** [apache-spark-on-k8s/spark#417]
This is a feature request to support multiple directories for spark.local.dir setup.
spark.local.dir use "/tmp" as the default setting ([link](https://spark.apache.org/docs/latest/configuration.html)). 
In Spark-on-Yarn, most customers usually use multiple directories for spark.local.dir setup. This way can help to get better performance.
In Spark-on-K8s, it indicates to root volume and cannot be modified by default. Since Spark-on-K8s running application and creating containers by request. This feature must create storage(PV) by request as well as configure directories in spark conf before launching the Spark applications.   

One related feature implementation(https://github.com/kubernetes/features/issues/121) from kubernetes.
We may need to wait for this feature implemented.

**Comment by ash211** [comment]
@weiting-chen do you need PV storage specifically, or would the `EmptyDir` from https://github.com/apache-spark-on-k8s/spark/pull/486 work for you?

I don't think you need persistence in static allocation mode, and dynamic allocation requires an external shuffle service which stores data in `spark.kubernetes.shuffle.dir`, not in `spark.local.dir`

**Comment by weiting-chen** [comment]
Yes, [#486](https://github.com/apache-spark-on-k8s/spark/pull/486) is enough for static mode.
Use PV storage doesn't make sense in spark.local.dir since the data is temporary and its life cycle comes with the executor pod.

**Comment by tangzhankun** [comment]
@ash211 @weiting-chen What determined the medium of emptydir by default?
